src/test/java/org/jenkinsci/plugins/credentialsbinding/impl/BindingStepTest.java

```diff
@@ -443,6 +440,80 @@ public void testTrackingOfCredential() {
         });
     }
 
+    @Test
+    public void userScopeCredentialsViaAuthorizeProject() {
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                enableAuthorizeProject(story.j);
+                User user = User.getOrCreateByIdOrFullName("aleph");
+                String credentialsId = UUID.randomUUID().toString();
+                String secret = "long passwords are great";
+                registerUserCredentials(user, new StringCredentialsImpl(CredentialsScope.USER, credentialsId,
+                        "User-scoped credential data from Aleph Null", Secret.fromString(secret)));
+
+                WorkflowJob p = story.j.createProject(WorkflowJob.class);
+                // run pipeline as aleph user
+                p.addProperty(new AuthorizeProjectProperty(new SpecificUsersAuthorizationStrategy("aleph", true)));
+                p.setDefinition(new CpsFlowDefinition(
+                        "node {\n" +
+                                "  withCredentials([string(variable: 'SECRET', credentialsId: '" + credentialsId + "')]) {\n" +
+                                "    writeFile file: 'secret.txt', text: env.SECRET\n" +
+                                "    def contents = readFile file: 'secret.txt'\n" +
+                                "    if (contents != '" + secret + "') error 'Could not access user-scoped credential'\n" +
+                                "  }\n" +
+                                "}",
+                        true));
+
+                story.j.buildAndAssertSuccess(p);
+            }
+        });
+    }
+
+    @Test
+    @Ignore("TODO: not implemented yet")
+    public void userScopeCredentialsViaBuildParameter() {
+        story.addStep(new Statement() {
+            @Override
+            public void evaluate() throws Throwable {
+                story.j.jenkins.setSecurityRealm(story.j.createDummySecurityRealm());
+                story.j.jenkins.setAuthorizationStrategy(new FullControlOnceLoggedInAuthorizationStrategy());
+                User user = User.getOrCreateByIdOrFullName("beth");
+                String credentialsId = UUID.randomUUID().toString();
+                String secret = "correct horse battery staple";
+                registerUserCredentials(user, new StringCredentialsImpl(CredentialsScope.USER, credentialsId,
+                        "User-scoped credential data from Beth Unit", Secret.fromString(secret)));
+
+                WorkflowJob p = story.j.createProject(WorkflowJob.class);
+                p.setDefinition(new CpsFlowDefinition(
+                        "node {\n" +
+                                "  error 'TODO: use a build parameter to prompt for credentials and use the above credentials'\n" +
```

Comment:
See [JENKINS-38963](https://issues.jenkins-ci.org/browse/JENKINS-38963) for a supposedly working syntax.

Reply:
Yes, the dollar sign syntax thing is explicitly coded for in BindingStep. I plan to use that syntax in a manual test of parameterized builds, though I also hope to figure out how to create an automated test around it.